rework_tests/unit/core/test_mixture.py

```diff
@@ -41,53 +45,102 @@ class TestMixtureModelInitialization:
         ],
     )
     def test_init_with_invalid_weights_raises_value_error(
-        self, exp_components: tuple[Exponential, Exponential], invalid_weights: list[float], error_msg: str
+        self, exp_components: tuple[Exponential, Exponential], invalid_weights: list[float], error_msg: str, dtype
     ):
         """Tests that initialization with invalid weights raises a ValueError."""
 
         with pytest.raises(ValueError, match=error_msg):
-            MixtureModel(components=exp_components, weights=invalid_weights)
+            MixtureModel(components=exp_components, weights=invalid_weights, dtype=dtype)
+
+    @pytest.mark.parametrize(
+        "invalid_weights, error_msg",
+        [
+            ([0.4, 0.4], "Sum of the weights must be equal 1, but it equal "),
+        ],
+    )
+    def test_init_with_invalid_sum_of_weights_raises_value_error(
+        self, exp_components: tuple[Exponential, Exponential], invalid_weights: list[float], error_msg: str, dtype
+    ):
+        """Tests that initialization with invalid sum of weights raises a ValueError."""
+        with pytest.raises(ValueError) as excinfo:
+            MixtureModel(components=exp_components, weights=invalid_weights, dtype=dtype)
 
-    def test_init_with_empty_components_raises_value_error(self):
+        actual_error_msg = str(excinfo.value)
+        assert actual_error_msg.startswith(error_msg)
+
+    def test_init_with_empty_components_raises_value_error(self, dtype):
         """Tests that initialization with an empty component list raises a ValueError."""
 
         with pytest.raises(ValueError, match="List of components cannot be an empty"):
-            MixtureModel(components=[])
+            MixtureModel(components=[], dtype=dtype)
+
+    def test_init_casts_component_dtypes(self, dtype):
+        """Tests that the MixtureModel casts all components to its own dtype during initialization."""
+
+        comp1_f64 = Exponential(loc=0.0, rate=1.0)
+        comp2_f64 = Exponential(loc=5.0, rate=2.0)
+        assert comp1_f64.dtype == np.float64
+
+        mixture = MixtureModel(components=[comp1_f64, comp2_f64], dtype=dtype)
+
+        assert mixture.dtype == dtype
+        assert mixture.weights.dtype == dtype
+        for component in mixture.components:
+            assert component.dtype == dtype
+            for param in component.params:
+                assert isinstance(getattr(component, param), dtype)
+
+        # Original components have not changed
+        for component in [comp1_f64, comp2_f64]:
+            assert component.dtype == np.float64
+            for param in component.params:
+                assert isinstance(getattr(component, param), np.float64)
+
+    def test_init_does_not_recreate_components_with_correct_dtype(self, dtype):
+        """Tests that components with the correct dtype are not recreated."""
+
+        comp_f32 = Exponential(loc=0.0, rate=1.0, dtype=dtype)
```

Comment:
Invalid variable name, here the component has arbitrary precision

```suggestion
        comp = Exponential(loc=0.0, rate=1.0, dtype=dtype)
```